docs/spec/distributing.rst

```diff
@@ -10,72 +10,428 @@ Stub files
-  or ``from .ham import Ham``, then ``ham`` is an exported attribute
-  of ``spam``.
+    def foo(): ...  # compatible
+    def bar(): pass  # behavior undefined
 
-* Stub files may be incomplete. To make type checkers aware of this, the file
-  can contain the following code::
+All variants of overloaded functions and methods must have an ``@overload``
+decorator::
 
-    def __getattr__(name) -> Any: ...
+    @overload
+    def foo(x: str) -> str: ...
+    @overload
+    def foo(x: float) -> int: ...
 
-  Any identifier not defined in the stub is therefore assumed to be of type
-  ``Any``.
+The following (which would be used in the implementation) is wrong in stubs::
+
+    @overload
+    def foo(x: str) -> str: ...
+    @overload
+    def foo(x: float) -> int: ...
+    def foo(x: str | float) -> Any: ...
+
+Aliases and NewType
+"""""""""""""""""""
+
+Type checkers should accept module-level type aliases, optionally using
+``TypeAlias`` (:pep:`613`), e.g.::
+
+  _IntList = list[int]
+  _StrList: TypeAlias = list[str]
+
+Type checkers should also accept regular module-level or class-level aliases,
+e.g.::
+
+  def a() -> None: ...
+  b = a
+
+  class C:
+      def f(self) -> int: ...
+      g = f
+
+A type alias may contain type variables. As per :pep:`484`,
+all type variables must be substituted when the alias is used::
+
+  _K = TypeVar("_K")
+  _V = TypeVar("_V")
+  _MyMap: TypeAlias = dict[str, dict[_K, _V]]
+
+  # either concrete types or other type variables can be substituted
+  def f(x: _MyMap[str, _V]) -> _V: ...
+  # explicitly substitute in Any rather than using a bare alias
+  def g(x: _MyMap[Any, Any]) -> Any: ...
+
+Otherwise, type variables in aliases follow the same rules as type variables in
+generic class definitions.
+
+``typing.NewType`` is also supported in stubs.
+
+Decorators
+""""""""""
+
+Type checkers are expected to understand the effects of all decorators defined
+in the ``typing`` module, plus these additional ones:
+
+ * ``classmethod``
+ * ``staticmethod``
+ * ``property`` (including ``.setter``)
+ * ``abc.abstractmethod``
+ * ``dataclasses.dataclass``
+ * functions decorated with ``@typing.dataclass_transform``
+
+The behavior of other decorators should instead be incorporated into the types.
+For example, for the following function::
+
+  import contextlib
+  @contextlib.contextmanager
+  def f():
+      yield 42
+
+the stub definition should be::
+
+  from contextlib import AbstractContextManager
+  def f() -> AbstractContextManager[int]: ...
+
+Version and Platform Checks
+"""""""""""""""""""""""""""
+
+Stub files for libraries that support multiple Python versions can use version
+checks to supply version-specific type hints. Stubs for different Python
+versions should still conform to the most recent supported Python version's
+syntax, as explain in the :ref:`stub-file-syntax` section above.
+
+Version checks are if-statements that use ``sys.version_info`` to determine the
```

Comment:
Version checking is already covered in other parts of the typing spec and shouldn't be duplicated here.